classes/eztagstreeattributefilter.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * eZTagsTreeAttributeFilter class implements TagsTreeAttributeFilter extended attribute
+ *
+ */
+class eZTagsTreeAttributeFilter
+{
+    /**
+     * Creates and returns SQL parts used in fetch functions
+     *
+     * @return array
+     */
+    function createSqlParts( $params )
+    {
+        $returnArray = array( 'tables' => '', 'joins'  => '', 'columns' => '' );
+        
+        if ( isset( $params['parent_tag_id'] ) )
+        {
+            if ( is_array( $params['parent_tag_id'] ) )
+            {
+                $tagIDArray = $params['parent_tag_id'];
+            }
+            else if ( is_string( $params['parent_tag_id'] ) )
+            {
+                $tagIDArray = array( $params['parent_tag_id'] );
+            }
+            else
+            {
+                return $returnArray;
+            }
+            $suffix = '/%';
+            if( isset( $params['exclude_parent'] ) && $params['exclude_parent'] )
+            {
+                $suffix = '/%/%';
+            }
+            $returnArray['tables'] = "  INNER JOIN eztags_attribute_link i1 ON (i1.object_id = ezcontentobject.id AND i1.objectattribute_version = ezcontentobject.current_version)
+                                        INNER JOIN eztags i2 ON (i1.keyword_id = i2.id)
+                                        INNER JOIN eztags_keyword i3 ON (i2.id = i3.keyword_id)";
+            $dbString = '( ';
+            $dbStringArray = array();
+            if( isset( $params['type'] ) && strtolower( $params['type'] ) == 'and' )
+            {
+                foreach( $tagIDArray as $tagID )
+                {
+                     $dbStringArray[] = "EXISTS (
+                                        SELECT 1
+                                        FROM
+                                            eztags_attribute_link j1,
+                                            ezcontentobject j2,
+                                            eztags j3
+                                        WHERE j3.path_string LIKE \"%/" . mysql_real_escape_string( $tagID . $suffix ) ."\"" .
```

Comment:
You should use `$db->escapeString()` here, due to support for PostgreSQL, or even better, now that $tagID needs to be numeric, in both of these `foreach` loops, fill the `$dbStringArray` only if `$tagID` is numeric and just cast it to int. For example: https://github.com/ezsystems/eztags/blob/master/classes/eztagsattributefilter.php#L121-L133